From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ishland <ishlandmc@yeah.net>
Date: Sun, 12 Dec 2021 17:19:00 -0500
Subject: [PATCH] (VMP) skip entity move if movement is zero

Code originally licensed under the MIT license and created for the VMP project: https://github.com/RelativityMC/VMP-fabric

diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 264871a01aef07a0691452bb6625a08aebbdf096..2d338839c02ca97e2c5b9de89beddee25195cf2d 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -951,6 +951,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
     // Paper end - detailed watchdog information
 
     public void move(MoverType movementType, Vec3 movement) {
+        if (movement.equals(Vec3.ZERO)) return; // JettPack - skip if movement is zero
         // Paper start - detailed watchdog information
         io.papermc.paper.util.TickThread.ensureTickThread("Cannot move an entity off-main");
         synchronized (this.posLock) {
